packages/fast-layouts-react/src/utilities/breakpoint-tracker.spec.ts

```diff
@@ -0,0 +1,68 @@
+import breakpointTracker, { Breakpoints, BreakpointTrackerCallback, defaultBreakpoints, identifyBreakpoint } from "./breakpoint-tracker";
+
+describe("identifyBreakpoint", (): void => {
+    test("should return the correct breakpoint values", (): void => {
+        expect(identifyBreakpoint(0, defaultBreakpoints)).toBe(0);
+        expect(identifyBreakpoint(539, defaultBreakpoints)).toBe(0);
+        expect(identifyBreakpoint(540, defaultBreakpoints)).toBe(1);
+        expect(identifyBreakpoint(767, defaultBreakpoints)).toBe(1);
+        expect(identifyBreakpoint(768, defaultBreakpoints)).toBe(2);
+        expect(identifyBreakpoint(1083, defaultBreakpoints)).toBe(2);
+        expect(identifyBreakpoint(1084, defaultBreakpoints)).toBe(3);
+        expect(identifyBreakpoint(1399, defaultBreakpoints)).toBe(3);
+        expect(identifyBreakpoint(1400, defaultBreakpoints)).toBe(4);
+        expect(identifyBreakpoint(1778, defaultBreakpoints)).toBe(4);
+        expect(identifyBreakpoint(1779, defaultBreakpoints)).toBe(5);
+    });
+});
+
+/* tslint:disable:no-string-literal */
+describe("breakpointTracker", (): void => {
+    let subscriber: any;
+
+    beforeEach(() => {
+        subscriber = {
+            onBreakpointChanged: (notification: BreakpointTrackerCallback): void => {
+                return;
+            }
+        };
+    });
+
+    test("should initialize automatically without the 'new' keyword", (): void => {
+        expect(breakpointTracker).toBeDefined();
+        expect(breakpointTracker).toBeTruthy();
+    });
+
+    test("should successfully track subscribers", (): void => {
```

Comment:
You actually shouldn't need unit tests for this. That "mocking is a code smell" article had some really useful insights for me for unit testing:

What you're really doing here is testing the internals of the tracker - which is the exact type of thing you *don't* want to test for. If I change up the way the internals this code works but *don't* change the interface with the code, then all of my tests should still pass because I haven't broken the API. If you test the internals though, then I need to update the code *and* the tests event though the API doesn't change - thus more work for no benefit.

A better way to test this is to add a subscription and ensure that the callback get's called when it is subscribed, and not get called after it is unsubscribed. Then you're testing subscription without probing the internal data structure of the component.